src/main/java/school/faang/user_service/service/RecommendationRequestService.java

```diff
@@ -0,0 +1,141 @@
+import org.springframework.transaction.annotation.Transactional;
+import school.faang.user_service.dto.RecommendationRequestDto;
+import school.faang.user_service.dto.RejectionDto;
+import school.faang.user_service.dto.RequestFilterDto;
+import school.faang.user_service.entity.RequestStatus;
+import school.faang.user_service.entity.Skill;
+import school.faang.user_service.entity.recommendation.RecommendationRequest;
+import school.faang.user_service.exception.DataValidationException;
+import school.faang.user_service.mapper.RecommendationMapper;
+import school.faang.user_service.repository.SkillRepository;
+import school.faang.user_service.repository.UserRepository;
+import school.faang.user_service.repository.recommendation.RecommendationRequestRepository;
+import school.faang.user_service.repository.recommendation.SkillRequestRepository;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
+public class RecommendationRequestService {
+    private final RecommendationRequestRepository recommendationRequestRepository;
+    private final RecommendationMapper recommendationMapper;
+    private final UserRepository userRepository;
+    private final SkillRepository skillRepository;
+    private final SkillRequestRepository skillRequestRepository;
+
+    public RecommendationRequestDto create(RecommendationRequestDto recommendationRequest) {
+        Long requesterId = recommendationRequest.getRequesterId();
+        Long receiverId = recommendationRequest.getReceiverId();
+
+        if (!userRepository.existsById(requesterId)) {
+            throw new EntityNotFoundException("Requester not found with id: " + requesterId);
+        }
+        if (!userRepository.existsById(receiverId)) {
+            throw new EntityNotFoundException("Receiver not found with id: " + receiverId);
+        }
+
+        Optional<RecommendationRequest> latestRequest = recommendationRequestRepository
+                .findLatestPendingRequest(requesterId, receiverId);
+
+        if (latestRequest.isPresent()) {
+            LocalDateTime sixMonthsAgo = LocalDateTime.now().minusMonths(6);
+            if (latestRequest.get().getCreatedAt().isAfter(sixMonthsAgo)) {
+                throw new DataValidationException("You can send a recommendation request to this user only once per 6 months");
+            }
+        }
+
+        if (recommendationRequest.getSkills() == null || recommendationRequest.getSkills().isEmpty()) {
+            throw new DataValidationException("Skills list must not be empty!");
+        }
+
+        recommendationRequest.getSkills().forEach(skillTitle -> {
+            if (!skillRepository.existsByTitle(skillTitle)) {
+                throw new DataValidationException("Skill not found: " + skillTitle);
+            }
+        });
+        RecommendationRequest newRequest = recommendationMapper.toEntity(recommendationRequest);
```

Comment:
Здесь получил ошибку - java.lang.NullPointerException: Cannot invoke "school.faang.user_service.entity.Skill.setTitle(String)" because the return value of "school.faang.user_service.entity.recommendation.SkillRequest.getSkill()" is null.

Похоже в маппере skill.getSkill() на 17й строчке null возвращает.
Лучше явно его инициализировать, напрмиер: